src/v/pandaproxy/rest/proxy.h

```diff
@@ -31,19 +32,25 @@ class proxy {
       const YAML::Node& config,
       ss::smp_service_group smp_sg,
       size_t max_memory,
-      ss::sharded<kafka::client::client>& client);
+      ss::sharded<kafka::client::client>& client,
+      sharded_client_cache& client_cache,
+      cluster::controller* controller);
 
     ss::future<> start();
     ss::future<> stop();
 
     configuration& config();
     kafka::client::configuration& client_config();
     ss::sharded<kafka::client::client>& client() { return _client; }
+    sharded_client_cache& client_cache() { return _client_cache; }
```

Comment:
This method should be added in the commit that calls `rq.service().client_cache()` in handlers.